### PR TITLE
feat: conda support | closes #3

### LINKED ISSUE #3: Conda support?
Would be great to have conda support. I'm trying to use this with neovide since neovide doesn't pickup the python env when it started up.

### PR DESCRIPTION
**Pull request by ChristianChiarulli**

### CODE REVIEW

File: lua/swenv/api.lua [hunk 1]
```diff
@@ -23,27 +44,43 @@ end
...
+    vim.notify('Could not require plenary: ' .. Path, vim.log.levels.WARN)
     return
   end
   local scan_dir = require('plenary.scandir').scan_dir
 
-  local paths = scan_dir(venvs_path, {depth = 1, only_dirs =true})
   local venvs = {}
+
+  -- CONDA
+  local conda_base_path = get_conda_base(vim.fn.getenv("CONDA_EXE"))
```
Comment (AckslD) [inline lua/swenv/api.lua]:
```suggestion
  local conda_base_path = get_conda_base(vim.fn.getenv("CONDA_EXE")) .. "/envs"
```
and remove the string joins below.

File: lua/swenv/api.lua [hunk 2]
```diff
@@ -6,13 +6,34 @@ local ORIGINAL_PATH = vim.fn.getenv('PATH')
 
 local current_venv = nil
 
+local get_conda_base = function(conda_exe_path)
+  local parts = {}
+
+  for part in conda_exe_path:gmatch("[^/]+") do
+    table.insert(parts, part)
+  end
+  table.remove(parts) -- remove the last part
```
Comment (AckslD) [inline lua/swenv/api.lua]: couldn't `parent` from `plenary.path` be used here?

File: lua/swenv/api.lua [hunk 3]
```diff
@@ -6,13 +6,34 @@ local ORIGINAL_PATH = vim.fn.getenv('PATH')
...
-  local venv_bin_path = venv.path .. '/bin'
-  vim.fn.setenv('PATH', venv_bin_path .. ':' .. ORIGINAL_PATH)
-  vim.fn.setenv('VIRTUAL_ENV', venv.path)
-  if settings.post_set_venv then
-    settings.post_set_venv(venv)
+  if venv.source == "conda" then
+    vim.fn.setenv('CONDA_PREFIX', venv.path)
+    vim.fn.setenv('CONDA_DEFAULT_ENV', venv.name)
+    vim.fn.setenv('CONDA_PROMPT_MODIFIER', "(" .. venv.name .. ")")
+    vim.cmd("LspRestart")
```
Comment (AckslD) [inline lua/swenv/api.lua]: Maybe this is indeed good to do but I wasn't sure if it had undesired consequences. Anyway, I think we should do it both for conda and venv or neither. Or would there be a reason to do it only for conda?

File: lua/swenv/api.lua [hunk 4]
```diff
@@ -6,13 +6,34 @@ local ORIGINAL_PATH = vim.fn.getenv('PATH')
...
+    -- TODO: remove old path
+    local conda_base_path = get_conda_base(vim.fn.getenv("CONDA_EXE"))
```
Comment (AckslD) [inline lua/swenv/api.lua]: Is this used?

File: lua/swenv/api.lua [hunk 5]
```diff
@@ -6,13 +6,34 @@ local ORIGINAL_PATH = vim.fn.getenv('PATH')
...
+    vim.fn.setenv('PATH', venv.path .. '/bin' .. ':' .. ORIGINAL_PATH)
```
Comment (AckslD) [inline lua/swenv/api.lua]: maybe we could move this to a common function also used below, eg `update_path` or similar?

**Review by AckslD — CHANGES_REQUESTED** [review]
Awesome thanks @ChristianChiarulli! This looks great, just added some comments. Also thanks for separating the formatting into its own commit, makes it much easier to review.

File: lua/swenv/api.lua (hunk 3 above)
Comment (ChristianChiarulli) [reply lua/swenv/api.lua]: I think it's necessary for the changes to take effect.

File: lua/swenv/api.lua (hunk 2 above)
Comment (ChristianChiarulli) [reply lua/swenv/api.lua]: yea, I wasn't aware of that. I updated it to use plenary, let me know if there's a better way to write it.

File: lua/swenv/api.lua [hunk 6]
```diff
@@ -23,27 +37,45 @@ end
...
+    vim.notify('Could not require plenary: ' .. Path, vim.log.levels.WARN)
     return
   end
   local scan_dir = require('plenary.scandir').scan_dir
 
-  local paths = scan_dir(venvs_path, {depth = 1, only_dirs =true})
   local venvs = {}
+
+  -- CONDA
+  local conda_env_path = Path.parent(Path.parent(Path:new(vim.fn.getenv('CONDA_EXE')))) .. '/envs'
```
Comment (AckslD) [inline lua/swenv/api.lua]: [nit] I find the following more readable but not super-important:
```suggestion
  local conda_env_path = Path:new(vim.fn.getenv('CONDA_EXE')):parent():parent() .. '/envs'
```

File: lua/swenv/api.lua [hunk 7]
```diff
@@ -6,11 +6,25 @@ local ORIGINAL_PATH = vim.fn.getenv('PATH')
...
 local set_venv = function(venv)
+  if venv.source == 'conda' then
+    vim.fn.setenv('CONDA_PREFIX', venv.path)
+    vim.fn.setenv('CONDA_DEFAULT_ENV', venv.name)
+    vim.fn.setenv('CONDA_PROMPT_MODIFIER', '(' .. venv.name .. ')')
+  else
+    vim.fn.setenv('VIRTUAL_ENV', venv.path)
+  end
+
+  vim.cmd('LspRestart')
```
Comment (AckslD) [inline lua/swenv/api.lua:22]: Sorry should have thought of this earlier but I think it's better to leave this up to the user. Very easy to do with the `post_set_venv` option, eg:
```lua
post_set_venv = function(venv) vim.cmd.LspRestart() end,
```

**Review by AckslD — CHANGES_REQUESTED** [review]

File: lua/swenv/api.lua (hunk 7 above)
Comment (ChristianChiarulli) [reply lua/swenv/api.lua:22]: sure, I'll add something in the readme about it

Comment (AckslD) [reply lua/swenv/api.lua:22]: Great! Then I think this is ready to go :)

**Comment by AckslD** [comment]
Thanks again for the contribution! I finished it of in #7, hope that was okay

**Pull request merged**